server/src/main/java/org/opensearch/extensions/ExtensionActionListener.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions;
+
+import java.util.ArrayList;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.admin.indices.analyze.AnalyzeAction.Response;
+
+/**
+ * ActionListener = for ExtensionsOrchestratore
+ *
+ * @opensearch.internal
+ */
+public class ExtensionActionListener<ExtensionBooleanResponse> implements ActionListener<Response> {
+
+    private static final Logger logger = LogManager.getLogger(ExtensionActionListener.class);
+    private ArrayList<Exception> exceptionList;
+
+    public ExtensionActionListener() {
+        exceptionList = new ArrayList<Exception>();
```

Comment:
Food for thought:  in a very long running setup, this list keeps getting added to and never emptied, will eventually consume all heap space. Consider a max capacity structure, like a circular FIFO queue, LRU queue, etc.

Don't hold up merging for this but consider adding an issue to discuss.

Reply:
https://github.com/opensearch-project/opensearch-sdk-java/issues/143

Reply:
+1